### DISCUSSION TITLE
Generic cosmetic filter doesn't get applied sometimes

### DISCUSSION THREAD

**Discussion by uBlock-user** [uBlockOrigin/uBlock-issues#3404] [General] [answered]
I face the issue(cosmetic filter not getting applied from the list) at `https://speed.cloudflare.com/`, once you complete the speedtest, a filter such as `##.react-share__ShareButton` doesn't get applied which is meant to hide the social media buttons on this site.

`##.react-share__ShareButton` exists in EasyList – Social Widgets.

Occurence is random, so cannot reproduce at will, but once it happens, I force update the list and it goes away for the time being.

This is how it looks like -- 

![Capture](https://github.com/user-attachments/assets/52c2b73c-6602-44d4-b049-4e42f03c64fd)

@gorhill You said something about `:watch-attr()`, can you expand upon that ?

**Comment by gorhill** [comment]
Try `*##.react-share__ShareButton` and see if this works better.

**Reply by uBlock-user** [reply]
It works, but are you able to reproduce the issue ? I want to know whats the reason behind this occurence.

**Reply by gorhill** [reply]
I wasn't able to reproduce, but you said it's random, so given how the page look (dynamically updated), most likely a case of requiring `watch-attr` -- .i.e. the `react-share__ShareButton` class is dynamically added after the element had already been surveyed element. Adding a class name or any attribute doesn't trigger the mutation observer, hence doesn't tell the surveyor to re-survey.

**Reply by uBlock-user** [reply]
> Adding a class name or any attribute doesn't trigger the mutation observer, 

Is this intentional due to performance issue related to m.o. ?

**Reply by gorhill** [reply, marked as answer]
Performance, it has been discussed in the issue tracker. There was one issue about a custom cosmetic filter on `arstechnica.com` suffering such issue, the poster thought it was caused by double dash `--`, but this was a attribute issue. I am having a hard time to find such old issues lately. I discussed that I was not comfortable with triggering the mutation observer on attribute changes. The solution is either inject uncondtionally (`*##`), or for procedural operator use `:watch-attr` operator.

**Reply by uBlock-user** [reply]
>  There was one issue about a custom cosmetic filter on arstechnica.com suffering such issue, the poster thought it was caused by double dash --, but this was a attribute issue. I am having a hard time to find such old issues lately.


Here - https://github.com/gorhill/uBlock/issues/1885
